### PR TITLE
Add TLS to the NIO implementation

### PR DESCRIPTION
**Pull request by glbrntt**
- Tests to follow in a separate PR
- Certificate and key from Echo have been updated since the certificate had expired
- EchoNIO uses the same certificate and key
- Also adds diagram to explain the server pipeline

### CODE REVIEW

File: Sources/SwiftGRPCNIO/GRPCServer.swift [hunk 1]
```diff
@@ -2,8 +2,79 @@ import Foundation
 import NIO
 import NIOHTTP1
 import NIOHTTP2
+import NIOSSL
 
 /// Wrapper object to manage the lifecycle of a gRPC server.
+///
+/// The pipeline is configured in three stages detailed below. Note: handlers marked with
+/// a '*' are responsible for handling errors.
```
Comment (weissi) [inline Sources/SwiftGRPCNIO/GRPCServer.swift]: 💜 the ascii art

**Comment by glbrntt** [comment]
I also submitted a PR against master with the updated Echo certificate and key (#407)

File: Sources/Examples/EchoNIO/main.swift [hunk 2]
```diff
@@ -29,11 +31,33 @@ let messageOption = Option("message",
                            default: "Testing 1 2 3",
                            description: "message to send")
 
+func makeClientTLSConfiguration() throws -> TLSConfiguration {
+  let certificate = try NIOSSLCertificate(file: "ssl.crt", format: .pem)
+  // The certificate common name is "example.com", so skip verification.
```
Comment (MrMage) [inline Sources/Examples/EchoNIO/main.swift]: nit: mind inserting "hostname" after "skip"?

**Comment by glbrntt** [comment]
> Great work, especially the flow chart! I do wonder, though, if we should provide a more convenient way of setting up "vanilla" SSL, especially on the client (see the gRPC-Core implementation, where you'd essentially just need to add `, secure: true`). Happy to have that in a later PR, though.

That's a good point. It's worth adding now since it should be fairly straightforward.

> When you add tests in a follow-up PR, consider doing it similar to the gRPC-Core implementation, where we essentially run all tests a second time with SSL enabled.

Exactly what I had in mind ;)

**Comment by rebello95** [comment]
@MrMage I trust your review - I'm not very familiar with the NIO implementation 😄

**Pull request merged**